### PR TITLE
Remove SubselectRewriter

### PR DESCRIPTION
**Pull request by mfussenegger**
See commit messages

### CODE REVIEW

**Comment by mfussenegger** [comment]
retest this please

File: sql/src/test/java/io/crate/planner/operators/LogicalPlannerTest.java [hunk 1]
```diff
@@ -240,6 +240,32 @@ public void testInWithSubqueryOrderImplicitlyAppliedWithExistingOrderByOnDiffere
...
+    public void testParentQueryIsPushedDownAndMergedIntoSubRelationWhereClause() {
+        LogicalPlan plan = plan("select * from " +
+                                " (select a, i from t1 order by a limit 5) t1 " +
+                                "left join" +
+                                " (select b, i from t2 where b > 10) t2 " +
+                                "on t1.i = t2.i where t1.a > 50 and t2.b > 100 " +
+                                "limit 10");
+        assertThat(plan, isPlan("FetchOrEval[a, i, b, i]\n" +
+                                "Limit[10;0]\n" +
+                                "HashJoin[\n" +
```
Comment (matriv) [inline sql/src/test/java/io/crate/planner/operators/LogicalPlannerTest.java]: Hash Join? with ``left`` :-)

Comment (mfussenegger) [reply sql/src/test/java/io/crate/planner/operators/LogicalPlannerTest.java]: Seems to be due to our left join -> inner join rewrite - I copied the query from a AnalyzerTest and didn't think about it. 

Can change it if you want.

Comment (mfussenegger) [reply sql/src/test/java/io/crate/planner/operators/LogicalPlannerTest.java]: @matriv pushed a fixup

**Comment by matriv** [comment]
bye bye SubSelectRewriter, you won't be missed...

**Pull request merged**